### DISCUSSION TITLE
Migration from Nextcloud AIO to a non AIO instance.

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#1416] [Questions] [answered]
How can this be done?
For me I have the following problem, I have a Nextcloud AIO instance and I want to switch to linking the containers myself ([https://github.com/nextcloud/all-in-one/tree/main/manual-install](https://github.com/nextcloud/all-in-one/tree/main/manual-install)).

**Comment by szaimen** [comment]
Do you want to do this switch on the same server?

**Reply by ghost** [reply]
no. i think that would make things more difficult the way i see it.

**Reply by szaimen** [reply]
Okay, so what is the reason then?

**Comment by szaimen** [comment]
Can you tell me more about the use case? Do e.g. you want to switch from one server that runs AIO to another that shall use the manual install?

**Reply by ghost** [reply]
Yes exactly, I currently have Nextcloud AIO running on Ubuntu server 22.04. Now I want to move the whole thing to TrueNAS (TrueNAS just has many advantages despite limiting GUI). The problem with TrueNAS there I have no way to access the docker.sock, which is why Nextcloud AIO unfortunately can not be run via TrueNAS. Sure you could run the whole thing in TrueNAS in a VM, but that's not really a clean solution.

TrueNAS has an official [Nextcloud app](https://nextcloud.com/blog/nextcloud-now-officially-supported-on-truenas/), but unfortunately Collabora, Turn, Redis, Clam, ... the Nextcloud app from TrueNAS can be linked to the respective required Docker containers. In addition, if you run multiple applications on a server and several for example need ClamAV (mail server, Nextcloud, ...) you can link TrueNAS and the Docker management the respective containers.

My plan was to backup everything with the backup app and restore everything in the new instance, but unfortunately I can't create a backup with the backup app. The button is grayed out for me.

**Reply by szaimen** [reply]
Do you mean TrueNas Core or TrueNas Scale?

**Reply by ghost** [reply]
Scale because of the Docker support

**Comment by szaimen** [comment]
I suppose easiest would be running AIO in a VM on TrueNas Scale then... You could import an AIO backup there then.

BTW: Since it is also not possible to use a docker-compose file on TrueNas Scale drectly IIRC, it also makes no sense to migrate to a manual instance.

**Reply by ghost** [reply]
But you can link the containers without Docker compose using enviromente variables afaik?

**Reply by ghost** [reply]
I just see --link is outdated but volumes can still be used to exchange data between containers.

**Reply by szaimen** [reply]
Yeah but you would need to translate the whole docker-compose into a truenas (kubernetes) config which is really complicated and not a solution I would recommend as you can make many mistakes by doing so. Also since the docker-compose file and containers might change in the future, you would always need to translate the latest changes as well...

**Reply by szaimen** [reply]
So if you want to keep using AIO, is a VM your best bet, IMHO.

**Reply by ghost** [reply]
Okay, yes, that's right, I also had my doubts about the solution and whether it was worth the effort and benefit.

**Reply by ghost** [reply]
However, you could take this with you and possibly make Nextcloud aware of it.... So that the official Nextcloud App at TrueNAS also redis and Turn and so contains because now times without these things a Nextcloud is rather cumbersome to use.

**Comment by szaimen** [comment, marked as answer]
See https://github.com/nextcloud/all-in-one#can-i-run-aio-on-truenas-scale
